addon/components/frost-password.js

```diff
@@ -1,103 +1,82 @@
 
-  // ==========================================================================
-  // Properties
-  // ==========================================================================
+  // == Properties ============================================================
 
-  classNames: ['frost-password'],
-  classNameBindings: ['revealable'],
+  classNames: [
+    'frost-password'
+  ],
+  classNameBindings: [
+    'isCapsOn',
+    'revealable'
+  ],
   isCapsOn: false,
-  isRevealerVisible: false,
+  isRevealed: false,
   layout,
   revealable: false,
-  revealed: false,
-  revealIcon: 'show',
   tabindex: 0,
-  type: 'password',
 
-  // ==========================================================================
-  // Computed Properties
-  // ==========================================================================
+  // == Computed properties  ===================================================
 
-  @readOnly
-  @computed('isCapsOn', 'isRevealerVisible', 'revealable')
-  isCapsAndReveal (isCapsOn, isRevealerVisible, revealable) {
-    return revealable && isCapsOn && isRevealerVisible
+  @computed('isRevealed')
+  revealMessage (isRevealed) {
+    return isRevealed ? 'Hide' : 'Show'
   },
 
-  // ==========================================================================
-  // Functions
-  // ==========================================================================
+  @computed('isRevealed')
+  type (isRevealed) {
+    return isRevealed ? 'text' : 'password'
+  },
 
-  // ==========================================================================
-  // Events
-  // ==========================================================================
+  // == Events ================================================================
 
-  focusOut: Ember.on('focusOut', function () {
+  _focusOut: on('focusOut', function (event) {
+    // this._super(...arguments)
     this.set('isCapsOn', false)
   }),
 
-  keyDown: Ember.on('keyDown', function (e) {
+  _keyDown: Ember.on('keyDown', function (e) {
     var s = e || window.e
-    if (this.get('isCapsOn') === false && (s.which === 20 || s.keyCode === 20)) {
+    if (s.which === 20 || s.keyCode === 20) {
       this.set('isCapsOn', true)
     }
   }),
 
-  keyUp: Ember.on('keyUp', function (e) {
-    var s = e || window.e
-    if (this.get('isCapsOn') === true && (s.which === 20 || s.keyCode === 20)) {
-      this.set('isCapsOn', false)
-    }
-  }),
-
-  keyPressed: Ember.on('keyPress', function (e) {
-    var s = String.fromCharCode(e.which || e.keyCode)   // IE support
-    if ((s.toUpperCase() === s && s.toLowerCase() !== s && !e.shiftKey) ||
-      (s.toUpperCase() !== s && s.toLowerCase() === s && e.shiftKey)) { // caps is on
+  _keyPress: Ember.on('keyPress', function (event) {
+    const c = String.fromCharCode(event.which || event.keyCode)
+    if ((c.toUpperCase() === c && c.toLowerCase() !== c && !event.shiftKey) ||
+      (c.toUpperCase() !== c && c.toLowerCase() === c && event.shiftKey)) {
```

Comment:
Also would it make sense to add `&& !this.get('isCapsOn')` to the `if` check so we don't set the property when it didn't actually change?